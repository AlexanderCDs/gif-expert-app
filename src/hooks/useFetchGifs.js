/**
 * @author: alexandercd
 * @date: 22/septiembre/2021
 * @description: 
**/
import { useState, useEffect } from 'react';
import{getGifs} from '../helpers/getGifs'; 

export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category, 10)
            .then( data => {
                setstate({
                    data,
                    loading: false
                })
            });
    }, [category])

    return state;
}

