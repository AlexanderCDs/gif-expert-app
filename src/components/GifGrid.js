/**
 * @author: alexandercd
 * @date: 22/septiembre/2021
 * @description: 
**/
import React from 'react'; 
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category, limit = 15}) => {

    const {data: images, loading} = useFetchGifs(category); 
 
    return (
        <>
            <h3>{category}</h3>
            <div className="container"> 
                {loading && (
                    <h3 className="animate__animated animate__flash">Loading...</h3>
                )}
                {images.map((item) => {
                    return (
                        <GifGridItem 
                            key={item.id} 
                            {...item }
                        />
                    )
                })}
            </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;