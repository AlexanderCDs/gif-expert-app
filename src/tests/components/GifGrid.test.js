/**
 * @author: alexandercd
 * @date: 24/septiembre/2021
 * @description: 
**/
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test <GifGrid />', () => {
    const category = 'back to the future'; 
    
    test('should show <GifGrid /> success', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });   
        const wrapper = shallow( <GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    }); 

    test('should show items when load images useFetchGif', () => {
        useFetchGifs.mockReturnValue({
            data: [{
                id: '12324',
                title: category,
                url: 'https://localhost/image.jpg'
            }],
            loading: false
        }); 
        const wrapper = shallow( <GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    });
    
    
});
