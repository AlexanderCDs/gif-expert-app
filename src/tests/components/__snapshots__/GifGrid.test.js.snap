// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should show <GifGrid /> success 1`] = `
<Fragment>
  <h3>
    back to the future
  </h3>
  <div
    className="container"
  >
    <h3
      className="animate__animated animate__flash"
    >
      Loading...
    </h3>
  </div>
</Fragment>
`;

exports[`Test <GifGrid /> should show items when load images useFetchGif 1`] = `
<Fragment>
  <h3>
    back to the future
  </h3>
  <div
    className="container"
  >
    <GifGridItem
      id="12324"
      key="12324"
      title="back to the future"
      url="https://localhost/image.jpg"
    />
  </div>
</Fragment>
`;
